### ISSUE TITLE
[BUG] `refine-react-router-v6` does not apply `options.route` correctly

### ISSUE THREAD

**Issue by rassie** [pankod/refine#1642]
**Describe the bug**

Given a configuration with two resources: 

```
                resources={[
                    {
                        name: 'users',
                        list: UserList
                    },
                    {
                        name: 'users',
                        list: LegacyUserList,
                        options: {
                            dataProviderName: 'users-legacy'
                            route: 'users-legacy',
                            label: 'Users (Legacy)'
                        },
                    }]
```

While both routes are available in the sense that they do not return 404 errors, the routing differs between versions.

With `refine-react-router`:

* `/users` renders `UserList`
* `/users-legacy` renders `LegacyUserList`

With `refine-react-router-v6`:

* `/users` renders `UserList`
* `/users-legacy` renders `UserList`

Also, `refine-react-router-v6` doesn't use the correct data provider, the rendering is exactly the same for `/users` and `/users-legacy`.

**Expected behavior**
`v6` should render the same as `v5`.

**Comment by omeraplak** [comment]
Hey @rassie ,
Thank you for contacting us! Now I created a CodeSandbox but I couldn't reproduce the situation you said. Is there a place I missed?
https://codesandbox.io/s/goy1dh

**Comment by rassie** [comment]
That's interesting, the problem is persistent locally. I'll try to figure out what's the difference. Thanks for testing, I'll re-open the issue if I find a culprit.

**Comment by rassie** [comment]
That's one of the things I'll have to check first, yes :)

**Comment by rassie** [comment]
Found it, the order of resources is important. See https://codesandbox.io/s/refine-base-example-forked-l5uxt7.

**Comment by omeraplak** [comment]
Hey @rassie ,
Thank you! We've released `@pankod/refine-react-router-v6@3.7.1`. Could you approve this fix?
